cmd/opentelemetry-collector/app/receiver/jaegerreceiver/jaeger_receiver.go

```diff
@@ -48,9 +51,58 @@ func (f *Factory) Type() configmodels.Type {
 func (f *Factory) CreateDefaultConfig() configmodels.Receiver {
 	cfg := f.Wrapped.CreateDefaultConfig().(*jaegerreceiver.Config)
 	cfg.RemoteSampling = createDefaultSamplingConfig(f.Viper)
+	configureAgent(f.Viper, cfg)
+	configureCollector(f.Viper, cfg)
 	return cfg
 }
 
+func configureAgent(v *viper.Viper, cfg *jaegerreceiver.Config) {
+	aOpts := agentApp.Builder{}
+	aOpts.InitFromViper(v)
+	if v.IsSet(thriftBinaryHostPort) {
+		cfg.Protocols["thrift_binary"] = &receiver.SecureReceiverSettings{
+			ReceiverSettings: configmodels.ReceiverSettings{
+				// TODO OTEL does not expose number of workers and queue length
+				Endpoint: v.GetString(thriftBinaryHostPort),
+			},
+		}
+	}
+	if v.IsSet(thriftCompactHostPort) {
+		cfg.Protocols["thrift_compact"] = &receiver.SecureReceiverSettings{
+			ReceiverSettings: configmodels.ReceiverSettings{
+				// TODO OTEL does not expose number of workers and queue length
```

Comment:
Wouldn't these be handled by the processors, e.g. queued_retry? So possibly they are not going to be supported by flags going forward - or possibly new flags that can be used to configure the processors?

Reply:
the workers and queue length are used on agent implementation it's agnostic to the queued_retry processor.